Cannot run test with swift-6.0-DEVELOPMENT-SNAPSHOT-2024-05-26-a

### Description

The [swift-6.0-DEVELOPMENT-SNAPSHOT-2024-05-26-a](https://github.com/apple/swift/releases/tag/swift-6.0-DEVELOPMENT-SNAPSHOT-2024-05-26-a), the latest 6.0 development snapshot toolchain, cannot run a test with swift-testing.

### Expected behavior

`swift test --disable-xctest` will succeed.

### Actual behavior

```console
$ swift test --disable-xctest

...

/path/to/swift-testing/Sources/Testing/Test.swift:127:22: error: conformance of 'AnySequence<Element>' to 'Sendable' is unavailable
125 |       preconditionFailure("Attempting to access test cases after evaluating them failed.")
126 |     }
127 |     return testCases.rawValue
    |                      `- error: conformance of 'AnySequence<Element>' to 'Sendable' is unavailable
128 |   }
129 | 

Swift.AnySequence:2:11: note: conformance of 'AnySequence<Element>' to 'Sendable' has been explicitly marked unavailable here
1 | @available(*, unavailable)
2 | extension AnySequence : Sendable {
  |           `- note: conformance of 'AnySequence<Element>' to 'Sendable' has been explicitly marked unavailable here
3 | }
[382/387] Compiling Testing Event.Symbol.swift
error: fatalError
```

### Steps to reproduce

1. Install [swift-6.0-DEVELOPMENT-SNAPSHOT-2024-05-26-a](https://github.com/apple/swift/releases/tag/swift-6.0-DEVELOPMENT-SNAPSHOT-2024-05-26-a) from https://www.swift.org/download/
2. `mkdir foo`
3. `cd foo`
4. `swift package init`
5. Open `Package.swift`, add `.package(url: "https://github.com/apple/swift-testing", from: "0.8.0")` to `dependencies`, and add `.product(name: "Testing", package: "swift-testing")` to `testTarget`'s `dependencies`
6. Run `swift test --disable-xctest`

### swift-testing version/commit hash

0.8.0 and main (578aa0cb6e518f0716dfec4448b5b9bd0d19d30d)

### Swift & OS version (output of `swift --version ; uname -a`)

```console
$ swift --version
Swift version 6.0-dev (LLVM 4b0b1f9e1a69523, Swift 91a3abcf816bc36)
Target: aarch64-unknown-linux-gnu
$ uname -a
Linux Brown-rhinoceros-beetle 6.8.9-406.asahi.fc40.aarch64+16k #1 SMP PREEMPT_DYNAMIC Wed May 22 14:40:24 UTC 2024 aarch64 GNU/Linux
```

--- a/Sources/Testing/Test.swift
+++ b/Sources/Testing/Test.swift
@@ -113,7 +113,7 @@ public struct Test: Sendable {
   /// combination of parameterized inputs. For non-parameterized tests, a single
   /// test case is synthesized. For test suite types (as opposed to test
   /// functions), the value of this property is `nil`.
-  var testCases: (some Sequence<Test.Case> & Sendable)? {
+  var testCases: (some Sequence<Test.Case>)? {
     testCasesState.map { testCasesState in
       guard case let .evaluated(testCases) = testCasesState else {
         // Callers are expected to first attempt to evaluate a test's cases by

